### PR TITLE
fix(itemstats): fix for backitem issue

### PR DESCRIPTION
**Pull request by DxCx**
Found an issue with applying stats over Backitem without details

### CODE REVIEW

**Comment by madou** [comment]
cool nice find!

i've actually added mocha into the project, will be adding tests at least for the more complicated parts). would you be keen to add a few so i don't accidentally break something down the track? :D

**Comment by DxCx** [comment]
cool, do you have istanbul (or nyc) as well? will be eaiser to make sure i'm covering everything..

**Comment by madou** [comment]
Ah! No.

Want to bring it in? : 👼

**Comment by DxCx** [comment]
I'll have time for it only next weekend but i don't mind helping out 👍

**Comment by DxCx** [comment]
Here is an example project i have it working:
https://github.com/DxCx/observable-diff-operator

**Comment by DxCx** [comment]
hey @madou 
i tried to do the integration to istanbul,
there are alot of issues with flow and babel, (never done it for none-typescript)

however, i've seen there is only 1 test at the moment,
if that's the case i'll suggest you to move to Jest which is a testing framework by Facebook.
it is built for React & flow usage.

checkout https://facebook.github.io/jest/docs/tutorial-react.html.

P.S
until coverage is working, for what change you want the unit test?

**Comment by madou** [comment]
I can setup coverage, I use it at work with a similar setup.

have you used mocha/proxyquire/chai/enzyme testing react apps before? if yes, whats your thoughts on it vs. jest? i've tried jest before but it was a little weird getting into it coming from mocha (especially the mocking...).

> P.S
> until coverage is working, for what change you want the unit test?

it would be good to get _some_ coverage over the stat item stuff since there's a decent amount of complicated logic !

**Comment by DxCx** [comment]
Hmm k

No i havent done nither react or es6 before lol
Im coming from typescript and angular :)
I wanted to learn react as well and so i did..
I have used jest before for typescript tho, its ver convinent that everything is handled under 1 tool
And if jest has built in support for react then why not?
Regarding proxyquire, didnt need to use it yet although i heard about it.
Most of the things im doing had dependcy injection (or di-like) so no need for patching

**Comment by madou** [comment]
Yeah outside of angular DI isn't really used. People just stub out import/requires with proxyquire

**Comment by madou** [comment]
Also fyi I found a little quirk with the current implementation of item stats.

If you want to have the same item appear in multiple item embeds with different stats, it won't work, for example:

```
<div
  data-armory-embed="items"
  data-armory-ids="79444"
  data-armory-79444-stat="1134"
>
</div>

<div
  data-armory-embed="items"
  data-armory-ids="79444"
  data-armory-79444-stat="1114"
>
</div>
```

It has a race condition with the `stat`, and whatever loads last will be what every item gets. I'm working towards fixing this - the fix being a new action/reducer for the `items` store that takes item/itemstat pairs, and then saves into the item store the generated item with stat (under `item:itemStat` key or something). This would allow us to reuse this for characters in the armory, too!

**Comment by DxCx** [comment]
Really, i knew it was not possible to have select twice on the same box, but what you showed here is indeed a bug. Nice catch!

**Comment by madou** [comment]
Hey dude, I've added coverage into the mix! Check it out.

**Comment by DxCx** [comment]
Nice! yeah i was able to add nyc, but then there was issue because it tried to instrument flow as well from some reason babel didn't strip it :(
then i got into a post which basiclly said, "common man, just use jest"

anyway,
i've added some basic testing even before the coverage, now i've rebased. =)

File: src/lib/gw2/itemStats.spec.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  name: 'Berserker\'s and Valkyrie',
+  attributes: {
+    Power: 0.35,
+    Precision: 0.25,
+    Vitality: 0,
+    CritDamage: 0.25,
+  },
+};
+
+const STAT_WIRED = {
```
Comment (madou) [inline src/lib/gw2/itemStats.spec.js]: STAT_WEIRD?

Comment (DxCx) [inline src/lib/gw2/itemStats.spec.js]: Invalid stat, something doesnt exists

Comment (madou) [inline src/lib/gw2/itemStats.spec.js]: yeah figured as much, all good!

merged and deployed to prod

fyi i brought in deploy to prod only for tagged commits, will make things a little easier from now on :)

**Pull request merged**